### ISSUE TITLE
oca-cla-bot fails with old commits and old emails

### ISSUE THREAD

**Issue by Yajo** [OCA/maintainer-tools#527]
## Describe the bug
oca-cla-bot fails when somebody migrates a module that includes commits from an email that had CLA signed back then, but not anymore.

## To Reproduce
See https://github.com/OCA/website/pull/882#issuecomment-1073983516

**Expected behavior**
My old job email should still pass ✔️ because when I pushed those commits back then, I had the CLA signed with that email.

**Comment by rousseldenis** [comment]
@Yajo I think this is tied to how we do migrations.

IMHO (I'll try to open a discussion soon), migrations should include only migration commits. It's becoming more and more difficult to review migrations.

We should create new main branches from preceding one.

**Comment by Yajo** [comment]
Well, it truly has a security problem, but outside of that I agree that the old process was even worse.

In any case, all of that is outside of the scope. The problem is that the bot complains about email addresses that were CLA-signed back then. It should compare the email *and date* of the commit, and then complain about those that have no CLA approval within that specific date.

**Comment by pedrobaeza** [comment]
Yes, I also put the same on https://github.com/OCA/oca-github-bot/issues/170. @sbidoul @gurneyalex maybe you can check it.

**Comment by sbidoul** [comment]
I don't think there is anything reasonable we can do to fix the current CLA bot. Long term we may want to move to a paperless and more robust process with, e.g., https://github.com/cla-assistant/cla-assistant.

**Comment by pedrobaeza** [comment]
@sbidoul we just need to include in the bot `with_context(active_test=False)` for searching inactive contacts

**Comment by gurneyalex** [comment]
I just added the context key in the legacy clabot. Let's see if it helps.
